src/network/core.rs

```diff
@@ -215,12 +215,12 @@ impl Core {
                         Err(err) => Err(err),
                     }
                 });
-                match handle.join() {
-                    Ok(interface_address) => interface_address,
+                match handle.join().unwrap() {
```

Comment:
Why use `unwrap()`. We want to handle errors properly and return them.

Reply:
@Luap99 If `join` fails at thread spawning level process will `panic` with appropriate message this is a special case. However the error returned by our task needs to  processed correctly which we are doing in `if` block.

Doc for `join` : https://doc.rust-lang.org/std/thread/struct.JoinHandle.html
```markdown
pub fn join(self) -> Result<T>

Waits for the associated thread to finish.

This function will return immediately if the associated thread has already finished.

In terms of atomic memory orderings, the completion of the associated thread synchronizes with this function returning. In other words, all operations performed by that thread happen before all operations that happen after join returns.

If the associated thread panics, Err is returned with the parameter given to panic!.
Panics

This function may panic on some platforms if a thread attempts to join itself or otherwise may create a deadlock with joining threads.
```

Reply:
nvm resetting this change for readability purpose. `{:?}` is needed for `join` since it returns `Boxed Any` which does not implements `display` but it will not affect wrapping the error.